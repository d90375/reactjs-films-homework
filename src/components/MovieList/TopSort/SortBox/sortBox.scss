.nav__format {
  display: flex;
  flex-direction: row;
  margin-top: 15px;

  .active {
    & div {
      background-color: #3bb4e0;
      opacity: 0.8;
    }
  }
}

.format__box4 {
  display: flex;
  flex-wrap: wrap;
  height: 16px;
  margin: 0 4px 0 0;
  width: 20px;
  padding: 0;
  border: 0;
  outline: none;
}

.format__box2 {
  border: 0;
  padding: 0;
  height: 16px;
  outline: none;
}

.box4 {
  background-color: #242b2d;
  border-radius: 2px;
  height: 7px;
  opacity: 0.15;
  margin-bottom: 2px;
  margin-right: 2px;
  width: 8px;
  cursor: pointer;

  &:last-child {
    margin-bottom: 0;
  }
}

.box2 {
  background-color: #242b2d;
  border-radius: 2px;
  height: 7px;
  opacity: 0.15;
  width: 17px;
  cursor: pointer;

  &:last-child {
    margin-top: 2px;
  }
}
