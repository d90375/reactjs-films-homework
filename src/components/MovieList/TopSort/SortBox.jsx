import React from "react";
import "./sortBox.scss";

const SortBox = () => {
  return (
    <div className="nav__format">
      <div className="format__box4">
        <div className="box4 box4_topLeft" />
        <div className="box4 box4_topRight" />
        <div className="box4 box4_btmLeft" />
        <div className="box4 box4_btmRight" />
      </div>
      <div className="format__box2">
        <div className="box2 box2_top" />
        <div className="box2 box2_btm" />
      </div>
    </div>
  );
};

export default SortBox;
