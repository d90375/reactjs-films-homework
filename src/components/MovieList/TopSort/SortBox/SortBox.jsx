import PropTypes from "prop-types";
import React from "react";
import "./sortBox.scss";

const SortBox = ({ isDisplayCardDirection, onSwitchToSquare, onSwitchToLine }) => {
  const squareStyle = isDisplayCardDirection === "square" ? `format__box4 active` : "format__box4";
  const squareLine = isDisplayCardDirection === "line" ? `format__box2 active` : "format__box2";

  return (
    <div className="nav__format">
      <button type="button" onClick={onSwitchToSquare} className={squareStyle}>
        <div className="box4 box4_topLeft" />
        <div className="box4 box4_topRight" />
        <div className="box4 box4_btmLeft" />
        <div className="box4 box4_btmRight" />
      </button>
      <button type="button" onClick={onSwitchToLine} className={squareLine}>
        <div className="box2 box2_top" />
        <div className="box2 box2_btm" />
      </button>
    </div>
  );
};

export default SortBox;

SortBox.propTypes = {
  isDisplayCardDirection: PropTypes.string,
  onSwitchToLine: PropTypes.func.isRequired,
  onSwitchToSquare: PropTypes.func.isRequired
};

SortBox.defaultProps = {
  isDisplayCardDirection: "square"
};
