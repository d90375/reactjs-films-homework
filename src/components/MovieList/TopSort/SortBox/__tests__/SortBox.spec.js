import React from "react";
import { shallow } from "enzyme";
import SortBox from "../SortBox";

describe("SortBox", () => {
  let component;

  const mockCallBacks = {
    onSwitchToSquare: jest.fn(),
    onSwitchToLine: jest.fn()
  };

  it("should render SortBox component without props", () => {
    component = shallow(<SortBox {...mockCallBacks} />);
    expect(component).toMatchSnapshot();
  });

  it("should render SortBox component with props", () => {
    component = shallow(<SortBox isDisplayCardDirection="line" {...mockCallBacks} />);
    expect(component).toMatchSnapshot();
  });

  describe("handlers", () => {
    it("should call onSwitchToSquare method", () => {
      expect(mockCallBacks.onSwitchToSquare.mock.calls.length).toBe(0);
      component.find(".format__box4").simulate("click");
      expect(mockCallBacks.onSwitchToSquare.mock.calls.length).toBe(1);
    });

    it("should call onSwitchToLine method", () => {
      expect(mockCallBacks.onSwitchToLine.mock.calls.length).toBe(0);
      component.find(".format__box2").simulate("click");
      expect(mockCallBacks.onSwitchToLine.mock.calls.length).toBe(1);
    });
  });
});
