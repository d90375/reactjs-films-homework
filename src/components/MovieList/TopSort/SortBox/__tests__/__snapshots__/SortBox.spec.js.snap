// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBox should render SortBox component with props 1`] = `
<div
  className="nav__format"
>
  <button
    className="format__box4"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="box4 box4_topLeft"
    />
    <div
      className="box4 box4_topRight"
    />
    <div
      className="box4 box4_btmLeft"
    />
    <div
      className="box4 box4_btmRight"
    />
  </button>
  <button
    className="format__box2 active"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="box2 box2_top"
    />
    <div
      className="box2 box2_btm"
    />
  </button>
</div>
`;

exports[`SortBox should render SortBox component without props 1`] = `
<div
  className="nav__format"
>
  <button
    className="format__box4 active"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="box4 box4_topLeft"
    />
    <div
      className="box4 box4_topRight"
    />
    <div
      className="box4 box4_btmLeft"
    />
    <div
      className="box4 box4_btmRight"
    />
  </button>
  <button
    className="format__box2"
    onClick={[MockFunction]}
    type="button"
  >
    <div
      className="box2 box2_top"
    />
    <div
      className="box2 box2_btm"
    />
  </button>
</div>
`;
